src/vpn-worker.ts

```diff
@@ -105,32 +105,45 @@ const worker = async (instanceId: number, serviceId: number) => {
 	});
 
 	const drainConnections = async () => {
-		const clientCount = Object.keys(clientCache).length;
+		let clientCount = Object.keys(clientCache).length;
 		logger.info(`${clientCount} clients connected`);
 
 		if (clientCount > 0) {
-			logger.debug(JSON.stringify(clientCache));
 			const delayMs = DEFAULT_SIGTERM_TIMEOUT / clientCount;
 			logger.info(
 				`disconnecting ${clientCount} clients, spaced by ${delayMs}ms`,
 			);
 
 			let timeToKill = DEFAULT_SIGTERM_TIMEOUT;
 
-			for (let clientId = 0; clientId < clientCount; clientId++) {
+			for (const [key] of Object.entries(clientCache)) {
```

Comment:
If you only want the key you're better off doing
```suggestion
			for (const key of Object.keys(clientCache)) {
```
but in this case I think you actually only care about the value, ie
```suggestion
			for (const cn of Object.values(clientCache)) {
```
and can remove all of the dancing around string vs number indexing below

Reply:
I assume you were using `Object.values` and then trying to use that to something that expected a string? I'm assuming it's fixed now because I would expect the current version to work?

Reply:
So far the only way I've been able to squelch these compiler errors is to use what Harrold suggested:
```
                       for (const [key] of Object.entries(clientCache)) {
                           try {
                               const _key = parseInt(key, 10);
```

Reply:
I think I got it..